easybuild/easyblocks/h/hadoop.py

```diff
@@ -93,6 +96,19 @@ def sanity_check_step(self):
         }
         super(EB_Hadoop, self).sanity_check_step(custom_paths=custom_paths)
 
+        fake_mod_data = self.load_fake_module(purge=True)
+        cmd = "hadoop checknative -a"
+        out, _ = run_cmd(cmd, log_all=True, simple=False, log_ok=True)
+        self.clean_up_fake_module(fake_mod_data)
+
+        not_found = []
+        for lib, lib_path  in self.cfg['extra_native_libs']:
+            if not re.search(r'%s: *true' % lib, out):
+                not_found.append(lib)
+        if not_found:
+            raise EasyBuildError("%s not found by 'hadoop checknative -a'" %
+                    ', '.join(not_found))
```

Comment:
no need to spread this across two lines? also, delay string interpolation until error actually happens:

```python
raise EasyBuildError("%s not found by 'hadoop checknative -a'", ', '.join(not_found))
```